### PR TITLE
Add support for environment based configuration

### PR DESCRIPTION
**Pull request by liviuignat**
- Added ability to override production config by setting an environment variable with a string JSON named `CONF`.
- Update Readme and Dockerfile
- add deploy task

### CODE REVIEW

**Comment by RaulTsc** [comment]
Will take a look later tonight 👍

**Comment by RaulTsc** [comment]
Added some comments. Other than that it looks fine to me 👍

File: config/production.js [hunk 1]
```diff
@@ -1,8 +1,39 @@
-module.exports = {
-  database: {
+const envConf = process.env.CONF;
+let prodConf = {};
 
-  },
+try {
+  prodConf = envConf ? JSON.parse(envConf) : { };
+} catch (err) {
+  console.log(`error starting up`);
```
Comment (liviuignat) [inline config/production.js]: yep

File: tasks/deploy [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+VERSION=$(docker run "$IMAGE_NAME:latest" node -e "console.log(require('./package.json').version)")
+docker build -t "$IMAGE_NAME:$VERSION" .
+
+echo "=====> PUSHING $IMAGE_NAME:$VERSION"
+VERSION=$(docker run "$IMAGE_NAME:latest" node -e "console.log(require('./package.json').version)")
+docker push "$IMAGE_NAME:$VERSION"
+
+echo "=====> PUSHING $IMAGE_NAME:$VERSION"
+VERSION=latest
+docker push "$IMAGE_NAME:$VERSION"
```
Comment (liviuignat) [inline tasks/deploy]: will improve this later.

**Pull request merged**